### ISSUE TITLE
微软或者 github 员工的开源项目可以考虑支持 996.icu 协议

### ISSUE THREAD

**Issue by zenglincient** [MSWorkers/support.996.ICU#141]

**Comment by Tedko** [comment]
公司的项目比较难，个人的其实可以考虑

**Comment by xtydev** [comment]
但用了就不能算是开源项目了

**Comment by xtydev** [comment]
# [Open Source Definition](https://opensource.org/osd)

**Comment by xtydev** [comment]
表示困惑的朋友可以参考下：
# [The Open Source Definition](https://opensource.org/osd)
特别是其中的第5条：
> ## 5. No Discrimination Against Persons or Groups
> The license must not discriminate against any person or group of persons.

以下是定义撰写人Bruce Perens对第5条的补充说明：
> _A license provided by the Regents of the University of California, Berkeley, prohibited an electronic design program from being used by the police of South Africa. While this was a laudable sentiment in the time of apartheid, it makes little sense today. Some people are still stuck with software that they acquired under that license, and their derived versions must carry the same restriction. Open Source licenses may not contain such provisions, no matter how laudable their intent._

然后我们再来看看[`反 996 许可证`](https://github.com/996icu/996.ICU/blob/master/LICENSE)中的第2条
> The individual or the legal entity must strictly comply with all applicable laws, regulations, rules and standards of the jurisdiction relating to labor and employment where the individual is physically located or where the individual was born or naturalized; or where the legal entity is registered or is operating (whichever is stricter). In case that the jurisdiction has no such laws, regulations, rules and standards or its laws, regulations, rules and standards are unenforceable, the individual or the legal entity are required to comply with Core International Labor Standards.

不难发现这个许可证存在两个潜在的问题：
1. 它的条款内容阻止了不遵守劳动法的公司使用该许可证下的软件或代码。虽然这正是该许可证的设计初衷，但也正因如此，它区别对待了“不遵守劳动法的公司”这一团体，与[OSD](https://opensource.org/osd)的第5条相悖。所以，[`反 996 许可证`](https://github.com/996icu/996.ICU/blob/master/LICENSE)是否能被归类为开源许可证还尚未明了（个人观点是：不能），与现有的开源许可证的兼容性也很难保证。
2. 现在的条款内容不能排除 Bruce Perens 引述的这种情况：我们无法保证所有国家/地区的劳动法永远是公平合理的。假设将来某个地区实行了压迫性的劳动法，那里良心尚在的实施公民抗命抵制恶法的公司反而会因许可证中的限制性条款被误伤。天真地认为劳动法总会站在劳动者这一边是非常短视的。

综上所述，[`反 996 许可证`](https://github.com/996icu/996.ICU/blob/master/LICENSE)的内容还有待完善。现阶段我并不建议任何个人和组织在项目中采用。请务必全面评估使用该许可证可能造成的影响。

**Comment by waynebaby** [comment]
如果改为 使用此授权下的ip 需要在项目中履行符合xxxx法规的 义务 就会好很多 因为是对未来行为的约束。可以这么理解吗？

**Comment by LeonXtp** [comment]
> 表示困惑的朋友可以参考下：
> 
> # [The Open Source Definition](https://opensource.org/osd)
> 特别是其中的第5条：
> 
> > ## 5. No Discrimination Against Persons or Groups
> > The license must not discriminate against any person or group of persons.
> 
> 以下是定义撰写人Bruce Perens对第5条的补充说明：
> 
> > _A license provided by the Regents of the University of California, Berkeley, prohibited an electronic design program from being used by the police of South Africa. While this was a laudable sentiment in the time of apartheid, it makes little sense today. Some people are still stuck with software that they acquired under that license, and their derived versions must carry the same restriction. Open Source licenses may not contain such provisions, no matter how laudable their intent._
> 
> 然后我们再来看看[`反 996 许可证`](https://github.com/996icu/996.ICU/blob/master/LICENSE)中的第2条
> 
> > The individual or the legal entity must strictly comply with all applicable laws, regulations, rules and standards of the jurisdiction relating to labor and employment where the individual is physically located or where the individual was born or naturalized; or where the legal entity is registered or is operating (whichever is stricter). In case that the jurisdiction has no such laws, regulations, rules and standards or its laws, regulations, rules and standards are unenforceable, the individual or the legal entity are required to comply with Core International Labor Standards.
> 
> 不难发现这个许可证存在两个潜在的问题：
> 
> 1. 它的条款内容阻止了不遵守劳动法的公司使用该许可证下的软件或代码。虽然这正是该许可证的设计初衷，但也正因如此，它区别对待了“不遵守劳动法的公司”这一团体，与[OSD](https://opensource.org/osd)的第5条相悖。所以，[`反 996 许可证`](https://github.com/996icu/996.ICU/blob/master/LICENSE)是否能被归类为开源许可证还尚未明了（个人观点是：不能），与现有的开源许可证的兼容性也很难保证。
> 2. 现在的条款内容不能排除 Bruce Perens 引述的这种情况：我们无法保证所有国家/地区的劳动法永远是公平合理的。假设将来某个地区实行了压迫性的劳动法，那里良心尚在的实施公民抗命抵制恶法的公司反而会因许可证中的限制性条款被误伤。天真地认为劳动法总会站在劳动者这一边是非常短视的。
> 
> 综上所述，[`反 996 许可证`](https://github.com/996icu/996.ICU/blob/master/LICENSE)的内容还有待完善。现阶段我并不建议任何个人和组织在项目中采用。请务必全面评估使用该许可证可能造成的影响。

那么此处应该@996icu

**Comment by xiechao06** [comment]
> A license provided by the Regents of the University of California, Berkeley, prohibited an electronic design program from being used by the police of South Africa

@xtydev 我不同意您的观点，首先osd第五款的***discriminate***是一个模糊的概念， 试问：限制罪犯的人身自由算不算***discrimination***? 我更赞成wikipedia上对[discrimination](https://en.wikipedia.org/wiki/Discrimination)的定义：

```
 Discrimination consists of treatment of an individual or group, based on their actual or perceived membership in a certain group or social category, "in a way that is worse than the way people are usually treated".
```

**区别对待**强制996的企业，并不是基于` their actual or perceived membership in a certain group or social category`, 	因此不能算是歧视。这些企业应该收到更严厉的惩罚，996.icu根本不能算是`worse than the way people are usually treated`.

另外在Bruce Perens的补充说明中，他使用了"may not contain such provisions"这样的词汇， 关于"may not"的定义，请参考[rfc2119](https://www.ietf.org/rfc/rfc2119.txt)。

一句话，我认为开源软件支持996.icu没有任何问题。 

===============================================================
For those who could not read Chinese:
===============================================================

[osd](https://opensource.org/osd-annotated) mentions: 

```
The license must not discriminate against any person or group of persons.
```

but what does "discriminate" mean?  Is "not granting" the copy of 996.icu licensed work to those companies violating labor laws a "discrimination"? Let's check wikipedia about [discrimination](https://en.wikipedia.org/wiki/Discrimination):

```
 Discrimination consists of treatment of an individual or group, based on their actual or perceived membership in a certain group or social category, "in a way that is worse than the way people are usually treated".
```

Treatment of those companies violating labor laws is not "based on their actual or perceived membership in a certain group or social category".

Besides, Bruce Perens did say "osd ... may not contain such provisions", but "may not" means "optional", you may read [rfc2119](https://www.ietf.org/rfc/rfc2119.txt).

**Comment by xtydev** [comment]
> Besides, Bruce Perens did say "osd ... may not contain such provisions", but "may not" means "optional", you may read [rfc2119](https://www.ietf.org/rfc/rfc2119.txt).

I see where you are coming from, but that's just plain **wrong**. RFCs may define Internet standards, but they are not very helpful in explaining the intricacies of the English language.

"**May not**" in the context of Bruce Perens's interpretation does not pertain to option. Here, it's all about **permission**, or the lack thereof. [This Q&A](https://english.stackexchange.com/questions/189974/why-do-they-say-may-not-for-things-which-people-shouldnt-do) on Stack Exchange might help you. Alternatively, just ask a native speaker 😉.

**Comment by xiechao06** [comment]
> > Besides, Bruce Perens did say "osd ... may not contain such provisions", but "may not" means "optional", you may read [rfc2119](https://www.ietf.org/rfc/rfc2119.txt).
> 
> I see where you are coming from, but that's just plain **wrong**. RFCs may define Internet standards, but they are not very helpful in explaining the intricacies of the English language.
> 
> "**May not**" in the context of Bruce Perens's interpretation does not pertain to option. Here, it's all about **permission**, or the lack thereof. [This Q&A](https://english.stackexchange.com/questions/189974/why-do-they-say-may-not-for-things-which-people-shouldnt-do) on Stack Exchange might help you. Alternatively, just ask a native speaker 😉.

Thank you for correcting me about "may not", seems that I've misunderstood "may not" for years 😭. But I think my main point is still true: 996.icu is not discrimination towards the organizations violating labor laws, and it does not violate OSD.

**Comment by xtydev** [comment]
> Thank you for correcting me about "may not,

😄 You're very much welcome!

> But I think my main point is still true: 996.icu is not discrimination towards the organizations violating labor laws, and it does not violate OSD.

WRT your main point, I'm still not convinced TBH. I do agree businesses that thrive on exploitation should be condemned and held accountable for the loss of their employees, but that is irrelevant because [Clause 5](https://opensource.org/osd#persons-or-groups) is not a moral judgement.

> Discrimination consists of treatment of an individual or group, based on their actual or perceived membership in a certain group or social category, "in a way that is worse than the way people are usually treated".

As to the definition of "discrimination" you've referenced, don't you think the terms of the Anti-996 License actually fit the description rather closely? In particular,
* We can clearly identify the group being discriminated against, namely, those who don't abide by the applicable labour-related regulations, rules or laws.
* We can tell exactly how they are being treated worse than others. That is, the terms of the license deny them the rights to deal with the licensed work for any purpose, while granting such rights to others.

The bottom line here is, the Anti-996 License effectively singles out those that don't abide by the applicable labour-related regulations, rules or laws and treats them differently, in violation of [Clause 5](https://opensource.org/osd#persons-or-groups) of the OSD. Therefore, it doesn't qualify as an open-source license, not in my book, anyway.

**Comment by buguang01** [comment]
感觉这个东西没有意义。

**Comment by kangzixiang** [comment]
Good
